src/averageAges.js

```diff
@@ -15,17 +15,33 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let counter = 0;
+  let diedInThisCentury;
+  let currentCentury = century;
+
+  const averageAge = people.reduce(function(prev, item) {
+    diedInThisCentury = Math.ceil(item.died / 100);
+
+    if (century === undefined) {
+      currentCentury = diedInThisCentury;
+    }
+
+    if (item.sex === 'm' && diedInThisCentury === currentCentury) {
+      counter++;
+
+      return prev + (item.died - item.born);
+    } else {
+      return prev;
+    }
```

Comment:
You can return prev without else

if (item.sex === 'm' && diedInThisCentury === currentCentury) {
      counter++;

  return prev + (item.died - item.born);
} 

  return prev;